libazureinit/src/error.rs

```diff
@@ -72,10 +75,181 @@ pub enum Error {
     Timeout,
     #[error("Failed to update the sshd configuration")]
     UpdateSshdConfig,
+    #[error("Failed to load sshd config: {details}")]
+    ConfigLoadFailure { details: String },
+    #[error("Unhandled exception")]
+    Unhandled { details: String },
 }
 
 impl From<tokio::time::error::Elapsed> for Error {
     fn from(_: tokio::time::error::Elapsed) -> Self {
         Self::Timeout
     }
 }
+
+/// Implement reportable formatting for `Error` to be used in health reporting.
+impl Error {
+    /// Returns a human-readable summary describing the error variant.
+    pub fn reason(&self) -> String {
```

Comment:
i'm inclined to ask that reason=lower case unless ACRONYM to be consistent with existing reasons used in cloud-init